Players-selection component

Agomik opened the pull request:
Put the "Plus" button at the end of shown players and ordered them by name.
I also translated in English some Italian text.

Commit:
Button's text is in English now

Commit:
Some strings translated in English.
Style of players view as in #40 ordered by name as requested in #33

Agomik committed:
Resolves #40 Create a basic component for selecting players.
Documentation and tests are to do and interoperability with player view has to be fixed.

Agomik committed:
players module controller and showplayers component tests are to do

astronati commented on src/app/components/show-players/show-players.view.html:
```diff
@@ -0,0 +1,19 @@
+        placeholder="add player"
+        ng-model="$ctrl.newPlayerName">
+    </form>
+</div>
```
Ogni riga deve finire con un `\n` per considerarsi tale. Ti allego questa domanda su stackoverflow che puo essere un'interessante lettura.
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Agomik replied:
Ho passato una vita a cancellare i \n che mette Gedit a fine file...interessante :)

astronati commented on src/app/components/show-players/show-players.component.js:
```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+function ShowPlayersController($element, SavedPlayers, SelectedPlayers, PlayersFactory) {
```
Per la definizione di un componente ti suggerirei il seguente approccio come gia fatto per gli altri file:
- definizione in un file
- template file esterno
- controller file esterno

astronati commented on src/app/modules/home/home.view.html:
```diff
@@ -14,7 +14,7 @@
        href="#!/players"
        role="button">
-      gioca
+      play
```
La traduzione in inglese fatta in questo modo non migliora in alcun modo l'applicazione. L'internazionalizzazione delle applicazioni prevede l'uso di token e di un servizio atto a intercettare la lingua usata dall'utente.

Agomik replied:
È meglio se uso per questo un servizio fatto da me o posso usare anche librerie come angular-translate?

astronati commented on src/app/modules/players/players.controller.js:
```diff
@@ -9,32 +9,27 @@ angular
        */
       vm.savedPlayers = SavedPlayers;
 
-      /**
```
Perche hai rimosso questi caratteri?

astronati commented on src/app/modules/players/players.controller.js:
```diff
@@ -150,4 +64,4 @@ angular
 
     }
-  ]);
+  ]);
```
Vedi commento sopra

astronati commented on src/app/modules/players/players.controller.spec.js:
```diff
@@ -1,11 +1,11 @@
 
 describe('Players controller', function() {
-  var createCtrl;
+//  var createCtrl;
```
Perche hai commentato il codice in questo modo? Immagino che non funzionava piu.
Qualche idea su come fixarlo piuttosto che commentarlo?

astronati commented on src/app/modules/players/players.view.html:
```diff
@@ -2,36 +2,20 @@
     <div ng-show="vm.savedPlayers.getAll().length">
       <!-- TODO this string needs to be translated -->
-      sono stati selezionati {{ vm.selectedPlayers.getAll().length }} giocatori
+      You have selected {{ vm.selectedPlayers.getAll().length }} player{{ vm.selectedPlayers.getAll().length != 1 ? "s" : "" }}
```
Uno dei tanti errori della traduzione in questo modo è che non permette di gestire il seguente caso:
- ITA `[n] giocatori sono stati selezionati`
- EN `You have selected [n] players`

Come risolveresti questo problema?

Agomik replied:
Considerando l'esistenza di un service per la traduzione direi di richiedere la stringa tramite una funzione che abbia come parametro il numero dei giocatori.
Come alternativa, attenendomi alla traduzione scelta (quindi senza fare "Players: [n]" :) ), ci potrebbe essere anche la delegazione al service della traduzione di leggere il numero dei giocatori selezionati ma non piace molto che il servizio di traduzione dipenda da selectedPlayers.

astronati commented on src/app/components/show-players/show-players.component.js:
```diff
@@ -0,0 +1,121 @@
+   * @description
+   * Visualizes created players names and icons.
+   */
+  .component('showPlayers', {
```
Qual è stata la scelta che ti ha fatto spostare tutta questa logica dal controller a questo component?
Poteva essere fatta in un modo migliore? Un componente che si chiama `showPlayers` fa esattamente cio che ti aspetti faccia?

astronati commented on src/app/components/show-players/show-players.view.html:
```diff
@@ -0,0 +1,19 @@
+<div class="adt-card"
+        ng-repeat="player in $ctrl.savedPlayers.getAll() | orderBy: 'name' as ordered_players track by player.id"
```
Non usiamo lo snake_case per le variabili ma il camelCase

Agomik committed:
Rinominato il component Show Players in Players Selection.
Il controller del componente è ora in un file esterno.
Test e documentazione significativi.
Aggiunto \n dove necessario.
Rimosso l'input form, ora l'aggiunta di un nuovo giocatore avviene con prompt() (un'alternativa più elegante sarebbe stata fare un dialog).
